pkg/resourcedefinitions/helper_test.go

```diff
@@ -41,24 +104,69 @@ func Test_alignSchemas(t *testing.T) {
 				givenOpenAPISchema.Components.Schemas == nil {
 				t.Fatalf("given file %q has no components.schemas", givenPath)
 			}
-			givenSchemas := givenOpenAPISchema.Components.Schemas
-			givens = append(givens, givenSchemas)
+			givenSchemas = givenOpenAPISchema.Components.Schemas
 		}
 
-		expectedPath := filepath.Join("testdata", tc.Name(), "expected.yaml")
-		expectedOpenAPISchema, err := loader.LoadFromFile(expectedPath)
-		if err != nil {
-			t.Fatalf("failed to load expected file %q: %v", expectedPath, err)
+		var expectedSchemas openapi3.Schemas
+		{
+			expectedPath := filepath.Join(dir, tc.Name(), "expected.yaml")
+			expectedOpenAPISchema, err := loader.LoadFromFile(expectedPath)
+			if err != nil {
+				t.Fatalf("failed to load expected file %q: %v", expectedPath, err)
+			}
+
+			if expectedOpenAPISchema.Components == nil ||
+				expectedOpenAPISchema.Components.Schemas == nil {
+				t.Fatalf("expected file %q has no components.schemas", expectedPath)
+			}
+
+			expectedSchemas = expectedOpenAPISchema.Components.Schemas
+		}
+
+		var nb map[string]any
+		{
+			nbJSONPath := filepath.Join(dir, tc.Name(), "nb.json")
```

Comment:
What does `nb.json` stand for?

Reply:
notebook, a shared data uses for the previous alignSchema func, here we mock that with this json file.

Reply:
Can we improve the naming or add comments

Reply:
comment in the testdata and the function.